### ISSUE TITLE
加载数据集卡住不动

### ISSUE THREAD

**Issue by airmanisvip** [ymcui/Chinese-LLaMA-Alpaca-2#309]
### 提交前必须检查以下项目

- [X] 请确保使用的是仓库最新代码（git pull），一些问题已被解决和修复。
- [X] 我已阅读[项目文档](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki)和[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/常见问题)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案。
- [X] 第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[LangChain](https://github.com/hwchase17/langchain)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)等，同时建议到对应的项目中查找解决方案。

### 问题类型

模型训练与精调

### 基础模型

Others

### 操作系统

Linux

### 详细描述问题

```
# 请在此处粘贴运行代码（请粘贴在本代码块里）
```
lr=2e-4
lora_rank=64
lora_alpha=128
lora_trainable="q_proj,v_proj,k_proj,o_proj,gate_proj,down_proj,up_proj"
modules_to_save="embed_tokens,lm_head"
lora_dropout=0.05

pretrained_model=/home/work/models
chinese_tokenizer_path=/home/work/models
dataset_dir=/home/work/dataset/dianping
data_cache=/home/work/dataset/dianping/cache
per_device_train_batch_size=1
gradient_accumulation_steps=8
block_size=512
output_dir=/home/liweijie/output

deepspeed_config_file=ds_zero2_no_offload.json

torchrun --nnodes 1 --nproc_per_node 1 run_clm_pt_with_peft.py \
    --deepspeed ${deepspeed_config_file} \
    --model_name_or_path ${pretrained_model} \
    --tokenizer_name_or_path ${chinese_tokenizer_path} \
    --dataset_dir ${dataset_dir} \
    --data_cache_dir ${data_cache} \
    --validation_split_percentage 0.001 \
    --per_device_train_batch_size ${per_device_train_batch_size} \
    --do_train \
    --seed 87464 \
    --fp16 \
    --num_train_epochs 1 \
    --lr_scheduler_type cosine \
    --learning_rate ${lr} \
    --warmup_ratio 0.05 \
    --weight_decay 0.01 \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --save_total_limit 3 \
    --save_steps 200 \
    --gradient_accumulation_steps ${gradient_accumulation_steps} \
    --preprocessing_num_workers 8 \
    --block_size ${block_size} \
    --output_dir ${output_dir} \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --lora_rank ${lora_rank} \
    --lora_alpha ${lora_alpha} \
    --trainable ${lora_trainable} \
    --lora_dropout ${lora_dropout} \
    --modules_to_save ${modules_to_save} \
    --torch_dtype float16 \
    --load_in_kbits 16 \
    --gradient_checkpointing \
    --ddp_find_unused_parameters False

运行后，代码一直卡在加载数据集那里 数据集是一个txt文件 显卡是RTX 4090 24G显存 即：
09/22/2023 20:25:55 - INFO - datasets.info - Loading Dataset Infos from /environment/miniconda3/lib/python3.10/site-packages/datasets/packaged_modules/text

### 依赖情况（代码类问题务必提供）

```
# 请在此处粘贴依赖情况（请粘贴在本代码块里）
```
(base) ➜  training git:(main) ✗ pip list | grep -E 'transformers|peft|torch|sentencepiece|bitsandbytes'
bitsandbytes                 0.41.0
peft                         0.5.0
sentencepiece                0.1.97
torch                        2.0.1+cu118
torchaudio                   2.0.2+cu118
torchvision                  0.15.2+cu118
transformers                 4.31.0

### 运行日志或截图

```
# 请在此处粘贴运行日志（请粘贴在本代码块里）
```
[INFO|tokenization_utils_base.py:1837] 2023-09-22 20:25:53,865 >> loading file tokenizer.model
[INFO|tokenization_utils_base.py:1837] 2023-09-22 20:25:53,865 >> loading file added_tokens.json
[INFO|tokenization_utils_base.py:1837] 2023-09-22 20:25:53,865 >> loading file special_tokens_map.json
[INFO|tokenization_utils_base.py:1837] 2023-09-22 20:25:53,865 >> loading file tokenizer_config.json
Using custom data configuration default-2addd6d49e256145
09/22/2023 20:25:55 - INFO - datasets.builder - Using custom data configuration default-2addd6d49e256145
Loading Dataset Infos from /environment/miniconda3/lib/python3.10/site-packages/datasets/packaged_modules/text
09/22/2023 20:25:55 - INFO - datasets.info - Loading Dataset Infos from /environment/miniconda3/lib/python3.10/site-packages/datasets/packaged_modules/text

**Comment by clclclaiggg** [comment]
> ### 提交前必须检查以下项目
> * [x] 请确保使用的是仓库最新代码（git pull），一些问题已被解决和修复。
> * [x] 我已阅读[项目文档](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki)和[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案。
> * [x] 第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[LangChain](https://github.com/hwchase17/langchain)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)等，同时建议到对应的项目中查找解决方案。
> 
> ### 问题类型
> 模型训练与精调
> 
> ### 基础模型
> 别人
> 
> ### 操作系统
> Linux目录
> 
> ### 详细描述问题
> ```
> # 请在此处粘贴运行代码（请粘贴在本代码块里）
> ```
> 
> lr=2e-4 lora_rank=64 lora_alpha=128 lora_trainable=“q_proj，v_proj，k_proj，o_proj，gate_proj，down_proj，up_proj” modules_to_save=“embed_tokens，lm_head” lora_dropout=0.05
> 
> pretrained_model=/家庭/工作/模型 chinese_tokenizer_path=/家庭/工作/模型 dataset_dir=/家庭/工作/数据集/大众点评 data_cache=/家/工作/数据集/大众点评/缓存 per_device_train_batch_size=1 gradient_accumulation_steps=8 block_size=512 output_dir=/首页/立维杰/输出
> 
> deepspeed_config_file=ds_zero2_no_offload.json
> 
> Torchrun --nnodes 1 --nproc_per_node 1 run_clm_pt_with_peft.py --deepspeed ${deepspeed_config_file} --model_name_or_path ${pretrained_model} --tokenizer_name_or_path ${chinese_tokenizer_path} --dataset_dir ${dataset_dir} --data_cache_dir ${data_cache} --validation_split_percentage 0.001 --per_device_train_batch_size ${per_device_train_batch_size} --do_train - -种子 87464 --FP16 --num_train_epochs 1 --lr_scheduler_type余弦 --learning_rate ${LR} --warmup_ratio 0.05 --weight_decay 0.01 --logging_strategy步 --logging_steps 10 --save_strategy步 --save_total_limit 3 --save_steps 200 --gradient_accumulation_steps ${gradient_accumulation_steps} --preprocessing_num_workers 8 --block_size ${block_size} --output_dir ${output_dir} --overwrite_output_dir --ddp_timeout 30000 --logging_first_step True --lora_rank ${lora_rank} --lora_alpha ${lora_alpha} --trainable ${lora_trainable} --lora_dropout ${lora_dropout} --modules_to_save ${modules_to_save} --torch_dtype float16 - -load_in_kbits 16 --gradient_checkpointing - -ddp_find_unused_parameters 假
> 
> 运行后，代码一直卡在加载数据集那里 数据集是一个txt文件 显卡是RTX 4090 24G显存 即： 09/22/2023 20：25：55 - 信息 - datasets.info - 从 /environment/miniconda3/lib/python3.10/site-packages/datasets/packaged_modules/text 加载数据集信息
> 
> ### 依赖情况（代码类问题务必提供）
> ```
> # 请在此处粘贴依赖情况（请粘贴在本代码块里）
> ```
> 
> （基础） ➜ 训练 git：（主） ✗ 点列表 |grep -E '变形金刚|火炬|句子|比特沙字节' 比特沙字节 0.41.0 佩夫特 0.5.0 句子 0.1.97 火炬 2.0.1+Cu118 火炬音频 2.0.2+CU118 火炬视觉 0.15.2+Cu118 变压器 4.31.0
> 
> ### 运行日志或截图
> ```
> # 请在此处粘贴运行日志（请粘贴在本代码块里）
> ```
> 
> [INFO|tokenization_utils_base.py：1837] 2023-09-22 20：25：53,865 >>加载文件标记器.model [INFO|tokenization_utils_base.py：1837] 2023-09-22 20：25：53,865 >>加载文件 added_tokens.json [INFO|tokenization_utils_base.py：1837] 2023-09-22 20：25：53,865 >>加载文件 special_tokens_map.json [INFO|tokenization_utils_base.py：1837] 2023-09-22 20：25：53,865 >>加载文件 tokenizer_config.json使用自定义数据配置 default-2addd6d49e256145 09/22/2023 20：25：55 - 信息 - datasets.builder - 使用自定义数据配置 default-2addd6d49e256145 从 /environment/miniconda3/lib/python3.10/site-packages/datasets/packaged_modules/text 加载数据集信息 09/22/2023 20：25：55 - 信息 - datasets.info - 从 /environment/miniconda3/lib/python3.10/site-packages/datasets/packaged_modules/text 加载数据集信息

你好，解决了吗，我也有同样的问题

**Comment by airmanisvip** [comment]
> > ### 提交前必须检查以下项目
> > 
> > * [x]  请确保使用的是仓库最新代码（git pull），一些问题已被解决和修复。
> > * [x]  我已阅读[项目文档](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki)和[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案。
> > * [x]  第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[LangChain](https://github.com/hwchase17/langchain)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)等，同时建议到对应的项目中查找解决方案。
> > 
> > ### 问题类型
> > 模型训练与精调
> > ### 基础模型
> > 别人
> > ### 操作系统
> > Linux目录
> > ### 详细描述问题
> > ```
> > # 请在此处粘贴运行代码（请粘贴在本代码块里）
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > lr=2e-4 lora_rank=64 lora_alpha=128 lora_trainable=“q_proj，v_proj，k_proj，o_proj，gate_proj，down_proj，up_proj” modules_to_save=“embed_tokens，lm_head” lora_dropout=0.05
> > pretrained_model=/家庭/工作/模型 chinese_tokenizer_path=/家庭/工作/模型 dataset_dir=/家庭/工作/数据集/大众点评 data_cache=/家/工作/数据集/大众点评/缓存 per_device_train_batch_size=1 gradient_accumulation_steps=8 block_size=512 output_dir=/首页/立维杰/输出
> > deepspeed_config_file=ds_zero2_no_offload.json
> > Torchrun --nnodes 1 --nproc_per_node 1 run_clm_pt_with_peft.py --deepspeed ${deepspeed_config_file} --model_name_or_path ${pretrained_model} --tokenizer_name_or_path ${chinese_tokenizer_path} --dataset_dir ${dataset_dir} --data_cache_dir ${data_cache} --validation_split_percentage 0.001 --per_device_train_batch_size ${per_device_train_batch_size} --do_train - -种子 87464 --FP16 --num_train_epochs 1 --lr_scheduler_type余弦 --learning_rate ${LR} --warmup_ratio 0.05 --weight_decay 0.01 --logging_strategy步 --logging_steps 10 --save_strategy步 --save_total_limit 3 --save_steps 200 --gradient_accumulation_steps ${gradient_accumulation_steps} --preprocessing_num_workers 8 --block_size ${block_size} --output_dir ${output_dir} --overwrite_output_dir --ddp_timeout 30000 --logging_first_step True --lora_rank ${lora_rank} --lora_alpha ${lora_alpha} --trainable ${lora_trainable} --lora_dropout ${lora_dropout} --modules_to_save ${modules_to_save} --torch_dtype float16 - -load_in_kbits 16 --gradient_checkpointing - -ddp_find_unused_parameters 假
> > 运行后，代码一直卡在加载数据集那里 数据集是一个txt文件 显卡是RTX 4090 24G显存 即： 09/22/2023 20：25：55 - 信息 - datasets.info - 从 /environment/miniconda3/lib/python3.10/site-packages/datasets/packaged_modules/text 加载数据集信息
> > ### 依赖情况（代码类问题务必提供）
> > ```
> > # 请在此处粘贴依赖情况（请粘贴在本代码块里）
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > （基础） ➜ 训练 git：（主） ✗ 点列表 |grep -E '变形金刚|火炬|句子|比特沙字节' 比特沙字节 0.41.0 佩夫特 0.5.0 句子 0.1.97 火炬 2.0.1+Cu118 火炬音频 2.0.2+CU118 火炬视觉 0.15.2+Cu118 变压器 4.31.0
> > ### 运行日志或截图
> > ```
> > # 请在此处粘贴运行日志（请粘贴在本代码块里）
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [INFO|tokenization_utils_base.py：1837] 2023-09-22 20：25：53,865 >>加载文件标记器.model [INFO|tokenization_utils_base.py：1837] 2023-09-22 20：25：53,865 >>加载文件 added_tokens.json [INFO|tokenization_utils_base.py：1837] 2023-09-22 20：25：53,865 >>加载文件 special_tokens_map.json [INFO|tokenization_utils_base.py：1837] 2023-09-22 20：25：53,865 >>加载文件 tokenizer_config.json使用自定义数据配置 default-2addd6d49e256145 09/22/2023 20：25：55 - 信息 - datasets.builder - 使用自定义数据配置 default-2addd6d49e256145 从 /environment/miniconda3/lib/python3.10/site-packages/datasets/packaged_modules/text 加载数据集信息 09/22/2023 20：25：55 - 信息 - datasets.info - 从 /environment/miniconda3/lib/python3.10/site-packages/datasets/packaged_modules/text 加载数据集信息
> 
> 你好，解决了吗，我也有同样的问题

没有，
